### ISSUE TITLE
Add --warmup_fraction to pretraining script

### ISSUE THREAD

**Issue by awaelchli** [Lightning-AI/litgpt#1234]
Luca suggested to add `pretrain --warmup_fraction 0.05` to define warmup in terms of a fraction of the epoch, which we are doing in one of our Studios. It's just something I decided on my own based on the paper referenced there, whether this is widely used to do warmup this way I don't know.

There were apparently users who tried running the code outside the Studio but it failed, so we now need to support this argument in LitGPT.

**Comment by rasbt** [comment]
That's not a bad idea and will be more robust for different dataset sizes. I wonder (and I don't know the answer) though if there is a minimum number of warm up step that should always be done, and a max number of steps that shouldn't be exceeded. 

For example if we use 0.05 and pretrain on 3T tokens, that's 150 billion warmup steps, which is a bit large :D

For reference, here are the number of steps for popular LLMs (taken from the [OLMo paper](https://arxiv.org/abs/2402.00838))

<img width="572" alt="Screenshot 2024-04-02 at 1 43 50 PM" src="https://github.com/Lightning-AI/litgpt/assets/5618407/e6be75a7-7a28-4b9e-88be-870e284b8e6b">

**Comment by awaelchli** [comment]
Yeah sorry for the confusion, this would only be used for continued pretraining. I forgot to mention that.
